contracts/sol6/helpers/KyberRateQueryReserves.sol

```diff
@@ -0,0 +1,161 @@
+pragma solidity 0.6.6;
+
+import "@kyber.network/utils-sc/contracts/Utils.sol";
+import "@kyber.network/utils-sc/contracts/IERC20Ext.sol";
+import "../IConversionRates.sol";
+import "../IKyberSanity.sol";
+import "../IKyberReserve.sol";
+
+interface IKyberReserveExt is IKyberReserve {
+    function conversionRatesContract() external view returns (IConversionRates);
+    function sanityRatesContract() external view returns (IKyberSanity);
+}
+
+contract KyberRateQueryReserves is Utils {
+
+    function getRatesWithEth(
+        IKyberReserveExt reserve,
+        IERC20Ext[] calldata tokens,
+        uint256 weiAmount
+    )
+        external view returns(uint256[] memory sellRates, uint256[] memory buyRates)
+    {
+        uint256 numTokens = tokens.length;
+
+        buyRates = new uint256[](numTokens);
+        sellRates = new uint256[](numTokens);
+        for (uint256 i = 0; i < numTokens; i++) {
+            (buyRates[i], sellRates[i], , , ) = getRateWithEth(reserve, tokens[i], weiAmount);
+        }
+    }
+
+    function getRatesWithToken(
+        IKyberReserveExt reserve,
+        IERC20Ext[] calldata tokens,
+        uint256 tweiAmount
+    )
+        external view 
+        returns(uint256[] memory sellRates, uint256[] memory buyRates)
+    {
+        uint256 numTokens = tokens.length;
+
+        buyRates = new uint256[](numTokens);
+        sellRates = new uint256[](numTokens);
+        for (uint256 i = 0; i < numTokens; i++) {
+            (buyRates[i], sellRates[i], ,  , ) = getRateWithToken(reserve, tokens[i], tweiAmount);
+        }
+    }
+
+    function getReserveRates(
+        IKyberReserveExt reserve,
+        IERC20Ext[] calldata srcs,
+        IERC20Ext[] calldata dests
+    )
+        external view returns(uint256[] memory pricingRates, uint256[] memory sanityRates)
+    {
+        require(srcs.length == dests.length, "srcs length != dests");
+
+        pricingRates = new uint256[](srcs.length);
+        sanityRates = new uint256[](srcs.length);
+        IKyberSanity sanityRateContract;
+        IConversionRates conversionRateContract;
+
+        try reserve.sanityRatesContract() returns (IKyberSanity sanityContract) {
+            sanityRateContract = sanityContract;
+        } catch {}
+
+        try reserve.conversionRatesContract() returns (IConversionRates ratesContract) {
+            conversionRateContract = ratesContract;
+        } catch {
+            revert("no conversionRate contract");
+        }
+
+        for (uint256 i = 0 ; i < srcs.length ; i++) {
+
+            if (reserve.sanityRatesContract() != IKyberSanity(0x0)) {
```

Comment:
we should use `sanityRateContract` at lines 75-76 and `conversionRateContract` at line 79 right?
Should also check line 97 and 134

Reply:
yes yes
Was being too lazy
see fix